[MPS] Slice matmul for large matrices

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117319
* #117223

Though, it does not work for large tensors right now, which makes me wonder if `-[sliceTensor:starts:ends:strides:]` is affected by the same malady

malfet committed:
[MPS] Slice matmul for large matrices

Though, it does not work for large tensors right now, which makes me wonder if `-[sliceTensor:starts:ends:strides:]` is affected by the same malady

[ghstack-poisoned]

malfet committed:
Update on "[MPS] Slice matmul for large matrices"


Though, it does not work for large tensors right now, which makes me wonder if `-[sliceTensor:starts:ends:strides:]` is affected by the same malady

[ghstack-poisoned]

albanD commented on aten/src/ATen/native/mps/operations/LinearAlgebra.mm:
```diff
@@ -28,15 +28,52 @@
+                                        name:nil];
+        auto chunkMM = [graph matrixMultiplicationWithPrimaryTensor:selfChunk secondaryTensor:otherChunk name:nil];
+
+        tile = tile ? [graph additionWithPrimaryTensor:tile secondaryTensor:chunkMM name:nil] : chunkMM;
```
Shouldn't the addition embed into the bigger Tensor here?

malfet replied:
MPSGraph is a unidirectional graph, that takes inputs and produces output, so there are no bigger tensor at this point, just a title.

albanD commented on aten/src/ATen/native/mps/operations/LinearAlgebra.mm:
```diff
@@ -28,15 +28,52 @@
 static std::tuple<MPSGraphTensor*, MPSGraphTensor*, MPSGraphTensor*> do_mm(MPSGraph* graph,
                                                                            const Tensor& self,
                                                                            const Tensor& other) {
+  constexpr auto tile_size = 16384;
```
A comment explaining why this value would be great.

malfet committed:
Update on "[MPS] Slice matmul for large matrices"


Though, it does not work for large tensors right now, which makes me wonder if `-[sliceTensor:starts:ends:strides:]` is affected by the same malady

[ghstack-poisoned]

malfet committed:
Update on "[MPS] Slice matmul for large matrices"


Though, it does not work for large tensors right now, which makes me wonder if `-[sliceTensor:starts:ends:strides:]` is affected by the same malady

[ghstack-poisoned]

malfet committed:
Update on "[MPS] Slice matmul for large matrices"


Though, it does not work for large tensors right now, which makes me wonder if `-[sliceTensor:starts:ends:strides:]` is affected by the same malady

[ghstack-poisoned]
